pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -49,7 +50,9 @@ import (
 )
 
 const finalizer = "flyte/flytek8s"
-
+const templateProjectVariable = "{{ PROJECT }}"
+const templateDomainVariable = "{{ DOMAIN }}"
```

Comment:
I think admin [here](https://github.com/flyteorg/flyteadmin/blob/76be8471380dbb6e58eeea4a6feb1601812eb9fb/pkg/clusterresource/controller.go#L61-L64) uses lower case to represent these variables... we should either keep this consistent or case insensitive.

I think you should leverage this [package instead](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/template/template.go) and just add project and domain as other fields in the Parameters struct... make them available to other templating we do throughout.

Reply:
Alright, thanks for the hint, wasn't aware of that package. I chose UPPERCASE to signalize that these are variables that will be filled by flyte itself unlike those in admin which I believe get filled with user-defined values that could have arbitrary keys. Can change to lowercase if you want though.

Reply:
I have the basic outline here https://github.com/bimtauer/flyteplugins/commit/9b6026ea6c47c6e9c7213302858537038572b64a but it looks like `Parameters` is very much tied to that file/s3 path use case. In this use-case now, all that is needed is passing the `TaskExecutionMetadata` to retrieve Project and Domain. But the other fields are not nullable. What to do here?

Reply:
To clarify: I'd like to be able to call `Render` with params simply being: 
```go
Parameters{
    TaskExecMetadata: taskMetadata,
}
```

Reply:
I removed marshalling now in favor of directly accessing the k,v pairs. Let me know what you think.